packages/cicero-core/src/templatelibrary.js

```diff
@@ -127,8 +134,15 @@ class TemplateLibrary {
             return Promise.resolve(result);
         }
 
+        if(this.httpHeader) {
+            Logger.info('Authenticated Template Index');
+        } else {
+            Logger.info('Template Index without Authentication and Authorization');
+        }
+
         const httpOptions = {
             uri: `${this.url}/template-library.json`,
+            httpHeader: this.httpHeader,
```

Comment:
`httpHeader` needs to be passed to the `Authorization` header in this object.

For example:
```js
        const httpOptions = {
            uri: `${this.url}/template-library.json`,
            headers: {
                'User-Agent': 'clause',
                Authorization: this.httpHeader,
            },
            json: true, // Automatically parses the JSON string in the response
        };
 ```